build_tools/hack/build_env_diag.py

```diff
@@ -18,1123 +19,2202 @@
+
+    def check_gxx(self):
+        gxx = self.device.gxx
+        gxx.description = "GCC Compiler C++ Language Frontend"
+
+        if self.device.LINUX:
+            if gxx.exe is None:
+                _stat = self.msg_stat("err", gxx, f"Cannot find {gxx.name} compiler.")
+                _except = cstring(
                     f"""
-    We found your MSVC install directory, but missing MSVC C/C++ compiler driver '{program}.exe'.
-    It seems your VC++ toolchain installation is broken. Please re-install {VISUAL_STUDIO_VERSION}.
-    traceback: No available {program} installed or {program} not in PATH
+    We can't find required C/C++ compilers.
+    On Linux platform we need GCC compilers to compile.
+    Please install it via package managers.
+        sh $ apt/dnf install gcc g++ binutils
+
+        traceback: GCC program {gxx.name} not installed or not in PATH
+           > Hint: Missing GNU {gxx.description}
     """,
-                    Color_Error,
-                )
-            program = "ml64".lower()
-            path = which(f"{VCToolsInstallDir}/bin/hostx64/x64/{program}.exe")
-            if path:
-                print(
-                    f"{Emoji_Pass} [{VISUAL_STUDIO_VERSION}/MSVC] Found MSVC Macro Assembler at: {path}."
+                    "err",
                 )
+                _result = None
+
             else:
-                print(
-                    f"{Emoji_Error} [{VISUAL_STUDIO_VERSION}/MSVC] Cannot find MSVC Macro Assembler."
+                _stat = self.msg_stat(
+                    "pass",
+                    gxx,
+                    f"Found GCC compiler program {gxx.name} {gxx.version} at {gxx.exe}",
                 )
-                printc(
-                    f"""
-    We found your MSVC install directory, but missing MSVC Macro Assembler '{program}.exe'.
-    This will effect on we compile TheRock sub-project [AMD-LLVM].
-    It seems your VC++ toolchain installation is broken. Please re-install {VISUAL_STUDIO_VERSION}.
+                _except = ""
+                _result = True
 
-    traceback: No available {program} installed or {program} not in PATH
+            return _stat, _except, _result
+
+    def check_gfortran(self):
+        gfortran = self.device.gfortran
+        gfortran.description = "GNU Fortran Compiler"
+
+        if gfortran.exe is None:
+            _stat = self.msg_stat(
+                "err", gfortran, f"Cannot found {gfortran.description}."
+            )
+            _except = cstring(
+                f"""
+    We cannot found any available {gfortran.description}.
+    On Windows, please install gfortran in your device, via Strawberry/MinGW-builds etc.
+    On Linux, please install via package managers.
+    Note: This requirement will be deprecated when TheRock team enables Flang/flang-new (LLVM based Fortran Compiler)
+        build on TheRock sub-project amd-llvm (ROCM/llvm-project).
+        traceback: No available Fortran compiler 'gfortran'
```

Comment:
Drop this note - no comments on future plans that are still up in the air, just focus on the current status.